File: src/main/java/de/cplaiz/activecraftcore/listener/VanillaCommandListener.java
package de.cplaiz.activecraftcore.listener;

import de.cplaiz.activecraftcore.messages.Errors;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerCommandPreprocessEvent;

public class VanillaCommandListener implements Listener {

    @EventHandler(priority = EventPriority.HIGHEST)
    public void onPlayerExecuteCommand(PlayerCommandPreprocessEvent event) {
        
        String message = event.getMessage().replace("/", "");

        String[] args;

        if (message.contains(" ")) {
            args = message.split(" ");
        } else {
            args = new String[1];
            args[0] = message;
        }

        if (args[0].equalsIgnoreCase("pl")  || args[0].equalsIgnoreCase("plugins") ||
                args[0].equalsIgnoreCase("bukkit:pl") || args[0].equalsIgnoreCase("bukkit:plugins")) {
            if(!event.getPlayer().hasPermission("activecraft.vanilla.plugins")) {
                event.getPlayer().sendMessage(Errors.NO_PERMISSION());
                event.setCancelled(true);
            }
        }

        if (args[0].equalsIgnoreCase("?") || args[0].equalsIgnoreCase("help") ||
                args[0].equalsIgnoreCase("bukkit:?") || args[0].equalsIgnoreCase("bukkit:help")) {
            if(!event.getPlayer().hasPermission("activecraft.vanilla.help")) {
                event.getPlayer().sendMessage(Errors.NO_PERMISSION());
                event.setCancelled(true);
            }
        }

        if (args[0].equalsIgnoreCase("icanhasbukkit") || args[0].equalsIgnoreCase("about") ||
                args[0].equalsIgnoreCase("bukkit:icanhasbukkit") || args[0].equalsIgnoreCase("bukkit:about")) {
            if(!event.getPlayer().hasPermission("activecraft.vanilla.about")) {
                event.getPlayer().sendMessage(Errors.NO_PERMISSION());
                event.setCancelled(true);
            }
        }

        if (args[0].equalsIgnoreCase("list") || args[0].equalsIgnoreCase("minecraft:list")) {
            if(!event.getPlayer().hasPermission("activecraft.vanilla.list")) {
                event.getPlayer().sendMessage(Errors.NO_PERMISSION());
                event.setCancelled(true);
            }
        }

        if (args[0].equalsIgnoreCase("me") || args[0].equalsIgnoreCase("minecraft:me")) {
            if(!event.getPlayer().hasPermission("activecraft.vanilla.me")) {
                event.getPlayer().sendMessage(Errors.NO_PERMISSION());
                event.setCancelled(true);
            }
        }

        if (args[0].equalsIgnoreCase("teammsg") || args[0].equalsIgnoreCase("tm") ||
                args[0].equalsIgnoreCase("minecraft:teammsg") || args[0].equalsIgnoreCase("minecraft:tm")) {
            if(!event.getPlayer().hasPermission("activecraft.vanilla.teammsg")) {
                event.getPlayer().sendMessage(Errors.NO_PERMISSION());
                event.setCancelled(true);
            }
        }

        if (args[0].equalsIgnoreCase("tell") || args[0].equalsIgnoreCase("w") ||
                args[0].equalsIgnoreCase("minecraft:tell") || args[0].equalsIgnoreCase("minecraft:w")) {
            if(!event.getPlayer().hasPermission("activecraft.vanilla.msg")) {
                event.getPlayer().sendMessage(Errors.NO_PERMISSION());
                event.setCancelled(true);
            }
        }

        if (args[0].equalsIgnoreCase("trigger") || args[0].equalsIgnoreCase("minecraft:trigger")) {
            if(!event.getPlayer().hasPermission("activecraft.vanilla.trigger")) {
                event.getPlayer().sendMessage(Errors.NO_PERMISSION());
                event.setCancelled(true);
            }
        }

        if (args[0].equalsIgnoreCase("ver") || args[0].equalsIgnoreCase("version") ||
                message.equalsIgnoreCase("bukkit:ver") || message.equalsIgnoreCase("bukkit:version")) {
            if(!event.getPlayer().hasPermission("activecraft.vanilla.version")) {
                event.getPlayer().sendMessage(Errors.NO_PERMISSION());
                event.setCancelled(true);
            }
        }
    }
}

File: src/main/java/org/activecraft/activecraftcore/listener/VanillaCommandListener.kt
package org.activecraft.activecraftcore.listener

import org.activecraft.activecraftcore.ActiveCraftCore
import org.activecraft.activecraftcore.playermanagement.Profile
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerCommandPreprocessEvent

class VanillaCommandListener : Listener {
    @EventHandler(priority = EventPriority.HIGHEST)
    fun onPlayerExecuteCommand(event: PlayerCommandPreprocessEvent) {
        val profile = Profile.of(event.player)
        val messageSupplier = profile.getMessageSupplier(ActiveCraftCore.INSTANCE)!!
        val message = event.message.replace("/", "")
        val args: Array<String?>
        if (message.contains(" ")) {
            args = message.split(" ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        } else {
            args = arrayOfNulls(1)
            args[0] = message
        }
        if (args[0].equals("pl", ignoreCase = true) || args[0].equals("plugins", ignoreCase = true) ||
            args[0].equals("bukkit:pl", ignoreCase = true) || args[0].equals("bukkit:plugins", ignoreCase = true)
        ) {
            if (!event.player.hasPermission("activecraft.vanilla.plugins")) {
                event.player.sendMessage(messageSupplier.errors.noPermission)
                event.isCancelled = true
            }
        }
        if (args[0].equals("?", ignoreCase = true) || args[0].equals("help", ignoreCase = true) ||
            args[0].equals("bukkit:?", ignoreCase = true) || args[0].equals("bukkit:help", ignoreCase = true)
        ) {
            if (!event.player.hasPermission("activecraft.vanilla.help")) {
                event.player.sendMessage(messageSupplier.errors.noPermission)
                event.isCancelled = true
            }
        }
        if (args[0].equals("icanhasbukkit", ignoreCase = true) || args[0].equals("about", ignoreCase = true) ||
            args[0].equals("bukkit:icanhasbukkit", ignoreCase = true) || args[0].equals(
                "bukkit:about",
                ignoreCase = true
            )
        ) {
            if (!event.player.hasPermission("activecraft.vanilla.about")) {
                event.player.sendMessage(messageSupplier.errors.noPermission)
                event.isCancelled = true
            }
        }
        if (args[0].equals("list", ignoreCase = true) || args[0].equals("minecraft:list", ignoreCase = true)) {
            if (!event.player.hasPermission("activecraft.vanilla.list")) {
                event.player.sendMessage(messageSupplier.errors.noPermission)
                event.isCancelled = true
            }
        }
        if (args[0].equals("me", ignoreCase = true) || args[0].equals("minecraft:me", ignoreCase = true)) {
            if (!event.player.hasPermission("activecraft.vanilla.me")) {
                event.player.sendMessage(messageSupplier.errors.noPermission)
                event.isCancelled = true
            }
        }
        if (args[0].equals("teammsg", ignoreCase = true) || args[0].equals("tm", ignoreCase = true) ||
            args[0].equals("minecraft:teammsg", ignoreCase = true) || args[0].equals("minecraft:tm", ignoreCase = true)
        ) {
            if (!event.player.hasPermission("activecraft.vanilla.teammsg")) {
                event.player.sendMessage(messageSupplier.errors.noPermission)
                event.isCancelled = true
            }
        }
        if (args[0].equals("tell", ignoreCase = true) || args[0].equals("w", ignoreCase = true) ||
            args[0].equals("minecraft:tell", ignoreCase = true) || args[0].equals("minecraft:w", ignoreCase = true)
        ) {
            if (!event.player.hasPermission("activecraft.vanilla.msg")) {
                event.player.sendMessage(messageSupplier.errors.noPermission)
                event.isCancelled = true
            }
        }
        if (args[0].equals("trigger", ignoreCase = true) || args[0].equals("minecraft:trigger", ignoreCase = true)) {
            if (!event.player.hasPermission("activecraft.vanilla.trigger")) {
                event.player.sendMessage(messageSupplier.errors.noPermission)
                event.isCancelled = true
            }
        }
        if (args[0].equals("ver", ignoreCase = true) || args[0].equals("version", ignoreCase = true) ||
            message.equals("bukkit:ver", ignoreCase = true) || message.equals("bukkit:version", ignoreCase = true)
        ) {
            if (!event.player.hasPermission("activecraft.vanilla.version")) {
                event.player.sendMessage(messageSupplier.errors.noPermission)
                event.isCancelled = true
            }
        }
    }
}